javalib/src/main/scala/java/util/stream/StreamImpl.scala

```diff
@@ -496,14 +516,51 @@ private[stream] class StreamImpl[T](
       coercedPriorStages
     )
       .asInstanceOf[DoubleStream]
+  }
+
+  def mapToInt(mapper: ToIntFunction[_ >: T]): IntStream = {
+    commenceOperation()
 
+    val spl = new Spliterators.AbstractIntSpliterator(
+      _spliter.estimateSize(),
+      _spliter.characteristics()
+    ) {
+      def tryAdvance(action: IntConsumer): Boolean =
+        _spliter.tryAdvance((e: T) => action.accept(mapper.applyAsInt(e)))
+    }
+
+    val coercedPriorStages = pipeline
+      .asInstanceOf[ArrayDeque[IntStreamImpl]]
+
+    new IntStreamImpl(
+      spl,
+      _parallel,
+      coercedPriorStages
+    )
+      .asInstanceOf[IntStream]
   }
 
-  def mapToInt(mapper: ToIntFunction[_ >: T]): IntStream =
-    throw new UnsupportedOperationException("Not Yet Implemented")
+  def mapToLong(mapper: ToLongFunction[_ >: T]): LongStream = {
+    commenceOperation()
+
+    val spl = new Spliterators.AbstractLongSpliterator(
+      _spliter.estimateSize(),
+      _spliter.characteristics()
+    ) {
+      def tryAdvance(action: LongConsumer): Boolean =
+        _spliter.tryAdvance((e: T) => action.accept(mapper.applyAsLong(e)))
+    }
+
+    val coercedPriorStages = pipeline
+      .asInstanceOf[ArrayDeque[LongStreamImpl]]
 
-  def mapToLong(mapper: ToLongFunction[_ >: T]): LongStream =
-    throw new UnsupportedOperationException("Not Yet Implemented")
+    new LongStreamImpl(
+      spl,
+      _parallel,
+      coercedPriorStages
+    )
+      .asInstanceOf[LongStream]
```

Comment:
This cast should not be required. It should be automatically widen to `LongStream` by compiler

Reply:
Fixed in several places in `java.util.stream`.    

The changes do make the code more idiomatic & consistent.